src/R2.jl

```diff
@@ -2,52 +2,66 @@ export R2, R2Solver
 
 """
     R2(nlp; kwargs...)
-    solver = R2Solver(nlp;)
+
+A first-order quadratic regularization method for unconstrained optimization.
+
+For advanced usage, first define a `R2Solver` to preallocate the memory used in the algorithm, and then call `solve!`.
+
+    solver = R2Solver(nlp)
     solve!(solver, nlp; kwargs...)
-    A first-order quadratic regularization method for unconstrained optimization
+
 # Arguments
 - `nlp::AbstractNLPModel{T, V}` is the model to solve, see `NLPModels.jl`.
+
 # Keyword arguments 
 - x0::V = nlp.meta.x0`: the initial guess
 - atol = eps(T)^(1 / 3): absolute tolerance
-- rtol = eps(T)^(1 / 3): relative tolerance: algorithm stop when ||∇f(x)|| ≤ ϵ_abs + ϵ_rel*||∇f(x0)||
+- rtol = eps(T)^(1 / 3): relative tolerance: algorithm stop when ||∇f(x)|| ≤ ϵ\\_abs + ϵ\\_rel*||∇f(x0)||
```

Comment:
```suggestion
- rtol = eps(T)^(1 / 3): relative tolerance: algorithm stops when ||∇f(x)|| ≤ ϵ\\_abs + ϵ\\_rel*||∇f(x0)||
```
Also, could you use ‖ instead of two | ?